changes/2020-07-14_multi-keyring-require-generation/change.md

```diff
@@ -0,0 +1,114 @@
+
+## Affected Implementations
+
+This serves as a reference for all implementations that this change affects.
+
+| Language   | Repository                                                                            |
+| ---------- | ------------------------------------------------------------------------------------- |
+| C          | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Javascript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+
+This change additionally affects the unreleased AWS KMS keyring implementations for Java and Python.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+The multi-keyring has a clearly configured intent of which keyring SHOULD be used to generate the data key,
+but this is not actually enforced during the encryption operation.
+This change strengthens the requirements on multi-keyrings such that if a generator is configured,
+it MUST generate the data key.
+
+## Out of Scope
+
+If a multi-keyring is configured with another multi-keyring as its generator,
+and this second multi-keyring has no generator,
+the outer multi-keyring cannot ever successfully complete an OnEncrypt operation.
+This is one of several examples of keyring configurations
+that cannot satisfy the requirements of an operation.
+Detecting at keyring configuration time that encryption/decryption will always fail
+because keyrings are not capable of meeting their requirements
+is [tracked separately](https://github.com/awslabs/aws-encryption-sdk-specification/issues/144).
+
+## Motivation
+
+If we do not make a guarantee that the generator will generate,
+then its usefulness is limited.
+Customers may have a compliance requirement that all data keys MUST be generated by a specific HSM,
+or may want to lock down true least-privilege permissions for their AWS KMS key policies.
+Under the current model,
+customers have to be very careful to make sure that they know
+exactly where and how a given multi-keyring configuration is being used
+rather than being certain that it will always do the same thing no matter what.
+
+## Drawbacks
+
+This is a breaking change,
+since existing multi-keyring configurations
+can be used in multiple contexts,
+and data key generation is not tightly controlled.
+We judge this to be the right trade-off
+between security and usability.
+
+## Security Implications
+
+Removing this flexibility may help identify potential security issues,
+as customers may discover when upgrading their ESDK version
+that data keys are not being generated as intended.
+Correcting this will allow customers
+to improve their security postures
+through improvements such as scoping down permissions.
```

Comment:
I would split this into two separate implications:

1. Do the correct thing.
1. Scope down permissions.